example-client/client.py

```diff
@@ -64,43 +64,76 @@ def run_structure(input: str) -> Optional[str]:
         args=[input],
     )
 
-    # Runs are asynchronous, so we need to poll the status until it's no longer running.
     structure_run_id = structure_run["structure_run_id"]
-    status = structure_run["status"]
-    printed_event_ids = set()  # Keep track of which events we've printed.
-    while status not in ("SUCCEEDED", "FAILED"):
-        structure_run = get_structure_run(
-            host=HOST, api_key=GT_API_KEY, run_id=structure_run_id
-        )
-        status = structure_run["status"]
 
-        # You can comment out this block if you don't want to stream events as they occur.
+    received_event_ids = set()  # Keep track of which events we've already received.
+    structure_has_terminal_event = False
+
+    # The client will continuously poll the Structure Run until it receives one of three events:
+    # 1. Our Structure has voluntarily emitted a FinishStructureRunEvent
+    # 2. The Structure has stopped running due to an error
+    # 3. The Structure has stopped running without an error
+    while not structure_has_terminal_event:
+        # TODO: Can the API be adjusted to give us all NEW events that occurred after a specific point?
```

Comment:
Can this be accomplished with pagination? Cloud will return pagination information, and the client can send subsequent ListEvent requests starting from the last page received

Reply:
Migrating [my comment](https://github.com/griptape-ai/managed-structure-template/pull/16/files#r1731864907).

Reply:
Can you outline pseudocode for how it will look for the client to do that? I don't want to jump to conclusions (sidebar: I already made that mistake when we were chatting in Slack about this).

Reply:
```python
    cursor = None  # Start with no cursor for the first page
    while True:
        page = fetch_page(cursor)  # Fetch a page of data
        process_data(page["data"])  # Process the data
        
        cursor = page["next_cursor"]  # Get the cursor for the next page
        if not cursor:  # If no more cursor, exit the loop
            break
```
@zachgiordano @vachillo does this type of pagination make sense to have in the Cloud API? I know we've already got offset based pagination for certain resources.

Reply:
That works for a single poll that has (potentially) multiple pages, but how do we maintain the cursor for the next poll iteration? What's the equivalent of "set the cursor to be just after the last page of events I pulled"?

Reply:
I'm not sure I follow..

We update the cursor on each poll iteration.
```python
cursor = page["next_cursor"]
```

Reply:
Yeah I think it makes sense. We already have pagination set up for [StructureRuns](https://github.com/griptape-ai/griptape-cloud/blob/main/control_plane/api/griptapecloud/components/structure_runs/views.py#L63-L65). 

It should be on Events as well. Seems we just haven't added it yet.